conkit/misc/selectalg.py

```diff
@@ -0,0 +1,172 @@
+        ess than the cutoff.
+
+        Parameters
+        ----------
+        data : list, tuple
+           A 1D array of scores
+        cutoff : float, optional
+           The cutoff of keeping decoys
+
+        Returns
+        -------
+        list
+           The decoy indices to keep
+        list
+           The decoy indices to throw
+
+        """
+        data = SubselectionAlgorithm._numpify(data)
+        keep = np.where(data >= cutoff)[0]
+        throw = np.where(data < cutoff)[0]
+        return keep.tolist(), throw.tolist()
+
+    @staticmethod
+    def linear(data, cutoff=0.5):
+        """A linearly-defined subselection algorithm
+
+        Description
+        -----------
+        This algorithm removes the worst 500 decoys.
+
+        Parameters
+        ----------
+        data : list, tuple
+           A 1D array of scores
+        cutoff : float, optional
+           The porportion of the total number of decoys to keep
+
+        Returns
+        -------
+        list
+           The decoy indices to keep
+        list
+           The decoy indices to throw
+
+        """
+        sorted_indices = SubselectionAlgorithm._numpify(data).argsort()[::-1]
+        point = np.ceil(sorted_indices.shape[0] * cutoff).astype(np.int)
+        keep = sorted_indices[:point]
+        throw = sorted_indices[point:]
+        return keep.tolist(), throw.tolist()
+
+    @staticmethod
+    def scaled(data, cutoff=0.5):
+        """A scaling-defined subselection algorithm
+
+        Description
+        -----------
+        This algorithm removes a decoy, if its scaled score
+        is less than 0.5. The scaled score is calculated by
+        dividing the precision score by the average of the
+        set.
+
+        Parameters
+        ----------
+        data : list, tuple
+           A 1D array of scores
+        cutoff : float, optional
+           The cutoff of keeping decoys
+
+        Returns
+        -------
+        list
+           The decoy indices to keep
+        list
+           The decoy indices to throw
+
+        """
+        data = SubselectionAlgorithm._numpify(data)
+        data_scaled = data / np.mean(data)
+        keep = np.where(data_scaled >= cutoff)[0]
+        throw = np.where(data_scaled < cutoff)[0]
+        return keep.tolist(), throw.tolist()
+
+    @staticmethod
+    def ignore(data):
+        """"A subselection algorithm to keep all
+
+        Description
+        -----------
+        This algorithm doesn't do anything except mimic others.
+
+        It will not discard any decoys and keep all!!
+
+        Parameters
+        ----------
+        data : list, tuple
+           A 1D array of scores
+
+        Returns
+        -------
+        list
+           The decoy indices to keep
+        list
+           The decoy indices to throw
+
+        """
+        data = SubselectionAlgorithm._numpify(data)
+        keep = np.where(data >= 0)[0]
+        throw = np.where(data < 0)[0]
```

Comment:
Again this could be replaced with a call to cutoff

Reply:
Good spot, thanks